### ISSUE TITLE
Node.js 0.10.40 compatibility

### ISSUE THREAD

**Issue by dick** [Azure/azure-iot-sdks#300]
I'm getting the errors below. Another commenter noted that an upgrade to Node 0.12 fixes it, but Meteor.js is fixed at 0.10.40 for now. Does anyone know of a workaround to get the azure-iothub module working with Node 0.10?

Thanks.

W20160213-13:04:05.432(-6)? (STDERR)          
W20160213-13:04:05.432(-6)? (STDERR) /Users/dick/.meteor/packages/meteor-tool/.1.1.10.1b51q9m++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/server-lib/node_modules/fibers/future.js:245
W20160213-13:04:05.432(-6)? (STDERR) 						throw(ex);
W20160213-13:04:05.432(-6)? (STDERR) 						      ^
W20160213-13:04:05.443(-6)? (STDERR) TypeError: Object prototype may only be an Object or null
W20160213-13:04:05.443(-6)? (STDERR)     at create (native)
W20160213-13:04:05.443(-6)? (STDERR)     at Function.create (/Users/dick/.meteor/packages/ecmascript-runtime/.0.2.6.1hlo1r4++os+web.browser+web.cordova/npm/node_modules/meteor-ecmascript-runtime/node_modules/core-js/modules/es6.symbol.js:87:60)
W20160213-13:04:05.443(-6)? (STDERR)     at Object.exports.inherits (util.js:556:27)
W20160213-13:04:05.444(-6)? (STDERR)     at Object.<anonymous> (/Users/dick/Dropbox/dev/petropi/.meteor/local/isopacks/npm-container/npm/node_modules/azure-iothub/node_modules/azure-iot-amqp-base/lib/amqp_receiver.js:47:6)
W20160213-13:04:05.444(-6)? (STDERR)     at Module._compile (module.js:456:26)
W20160213-13:04:05.444(-6)? (STDERR)     at Object.Module._extensions..js (module.js:474:10)
W20160213-13:04:05.444(-6)? (STDERR)     at Module.load (module.js:356:32)
W20160213-13:04:05.444(-6)? (STDERR)     at Function.Module._load (module.js:312:12)
W20160213-13:04:05.444(-6)? (STDERR)     at Module.require (module.js:364:17)
W20160213-13:04:05.444(-6)? (STDERR)     at require (module.js:380:17)
W20160213-13:04:05.444(-6)? (STDERR)     at Object.<anonymous> (/Users/dick/Dropbox/dev/petropi/.meteor/local/isopacks/npm-container/npm/node_modules/azure-iothub/node_modules/azure-iot-amqp-base/lib/amqp.js:8:20)
W20160213-13:04:05.444(-6)? (STDERR)     at Module._compile (module.js:456:26)
W20160213-13:04:05.444(-6)? (STDERR)     at Object.Module._extensions..js (module.js:474:10)
W20160213-13:04:05.445(-6)? (STDERR)     at Module.load (module.js:356:32)
W20160213-13:04:05.445(-6)? (STDERR)     at Function.Module._load (module.js:312:12)
W20160213-13:04:05.445(-6)? (STDERR)     at Module.require (module.js:364:17)

**Comment by pierreca** [comment]
Hi @dick
I quickly looked into this and it looks like it might be linked to changes in the way the EventEmitter module exported its main EventEmitter constructor.

in the files that you need that inherit from EventEmitter, try changing the EventEmitter require statement to:
```js
var EventEmitter = require('events').EventEmitter
```
Please let us know how this turns out.  It's unlinkely that this is going to be the only change required to make the SDK work on Node 0.10 but if we get to a solution that makes the SDK works with it without breaking compatibility with the more recent version, I'll be delighted to accept a PR or implement the changes myself.

**Comment by dick** [comment]
Hi Pierre,

I gave it a brief shot, but updating the require statements didn't get it
working. I appreciate you taking a look, but for now I'll just move over to
using the Rest API.

**Comment by dick** [comment]
Update: I seem to have gotten the sdk working with Meteor.js, but I'm not knowledgable enough to fully explain it, and the change I made is specific to Meteor. I *think* the issue is with requiring Events and Utilities because they are not explicitly listed as a dependency because they are internal Node modules. This seems to be a Meteor.js npm issue, not an issue with the Node.js version as I originally thought. 

A dependency for my hack is using the meteorhacks npm integration here: https://github.com/meteorhacks/npm

In amqp_receiver.js, I then changed the first two require statements as follows:
`var EventEmitter = Meteor.npmRequire('events').EventEmitter;
var util = Meteor.npmRequire('util');`

Using a console I was able to add a new device and send a cloud to message device without issue. If I end up seeing any more errors with this approach I will post them here.
